runtime/onert/core/src/compiler/ExecutorFactory.cc

```diff
@@ -201,18 +201,35 @@ ExecutorFactory::initializeModelIOTensors(ir::LoweredGraph &lowered_graph,
     // Add tensor to controlflow TensorRegistry.
     cf_tensor_builder->setUserTensor(ind, tensor);
     ret.push_back(tensor);
-
-    // Set other tensors as external tensors
-    for (auto &tensor_builder : tensor_builders)
-    {
-      // FIXME This is a workaround registering all user tensors to all backends
-      // FIXME Handle when it is failed
-      tensor_builder->setExternalTensor(ind, tensor);
-    }
   }
   return ret;
 }
 
+void ExecutorFactory::prepareExternalTensors(ir::LoweredGraph &lowered_graph,
+                                             TensorBuilders &tensor_builders)
+{
+  lowered_graph.iterateTopolOpSeqs(
```

Comment:
Does this method require topological sort of `OpSequences`?

Reply:
I think not.

Reply:
If so, you can use `op_seqs()` of `LoweredGraph`  instead of `iterateTopolOpSeqs()`

Reply:
Ok thanks!